apps/src/blockly/addons/cdoBlockSerializer.js

```diff
@@ -13,7 +31,12 @@ export default class CdoBlockSerializer extends GoogleBlockly.serialization
    * @param {Blockly.Workspace} workspace - The workspace to deserialize into.
    */
   load(stateToLoad, workspace) {
-    const blockStates = stateToLoad['blocks'];
+    // Procedure definitions should be loaded ahead of call blocks, so that the
+    // procedures map is updated correctly.
+    const blockStates = sortBlocksByType(
+      stateToLoad['blocks'],
+      procedureDefinitionTypes
```

Comment:
nit: If this is something that's not really meant to be modified (or modified with great caution), should we make it a capital PROCEDURE_DEFINITION_TYPES? And I like that your argument name in `sortBlocksByType` -- `prioritizedBlockTypes` -- should we modify the name of this constant to indicate that they reflect priority?

Reply:
Good call on the capitalization to notate that it shouldn't be modified.

I intentionally used `prioritizedBlockTypes` for the parameter name because that's how that argument is used in the context of the function. 

The list we're importing (now called `PROCEDURE_DEFINITION_TYPES`) is simply some strings that signify block types. I imagine that it could be used for other things (such as determining which blocks need to be moved off the main workspace during serialization).

While it's less likely, I also feel like one could find a reason to pass a different list into the sort function. For example, we might one day decide to sort our blocks so that blocks of type `"when_run"` are sorted first before we position blocks on the workspace.

I was trying to keep the functionality of the sorts clear through parameter naming, while the name of the value being passed is more to indicate how we're using the sort.

Let me know your thoughts!

Reply:
Maybe something like:

```
  const prioritizedBlocks = blockStates.filter((block) =>
    prioritizedBlockTypes.includes(block.type)
  );

  const remainingBlocks = blockStates.filter((block) =>
    !prioritizedBlockTypes.includes(block.type)
  );

  return [...prioritizedBlocks, ...remainingBlocks];
```

Reply:
I like that, and I think the two conversations we're having are coming full circle. I proposed something very similar here: https://github.com/code-dot-org/code-dot-org/pull/52932/files#r1277954940